src/xkbcomp/rules.c

```diff
@@ -988,8 +988,45 @@ expand_rmlvo_in_kccgst_value(struct matcher *m, struct scanner *s,
 }
 
 /*
- * This function performs %-expansion on @value (see overview above),
- * and appends the result to @to.
+ * This function performs :all replacement on @value (see overview above),
+ * and appends the result to @expanded.
+ */
+static bool
+expand_qualifier_in_kccgst_value(
+    struct matcher *m, struct scanner *s,
+    struct sval value, darray_char *expanded, unsigned int *i, unsigned int last_item_idx)
+{
+    const char *str = value.start;
+
+    if (*i + 2 >= value.len || str[*i] != 'a' || str[*i+1] != 'l' || str[*i+2] != 'l') {
+        /* Not :all qualifier */
+        return true;
+    }
+    /* Found :all, check leftovers until the end of the qualifier */
+    if (*i + 3 <= value.len || str[*i + 3] == '+' || str[*i + 3] == '|') {
+        /* Add at least one layout */
```

Comment:
Why?

Reply:
This checks we have _exactly_ `all`. I refactored wrt to your other comment about positive test.